### PR TITLE
Create experimental hot-swap dds package

### PR DESCRIPTION
**Pull request by tyler-cai-microsoft**
[AB#5740](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/5740)

## Description

Here is the parent PR: https://github.com/microsoft/FluidFramework/pull/17564

The goal is to create an experimental package for the hotswap DDS that we do not currently wish publish to **npm**.
Notes:
- This is so internal development until enough confidence has been created for customer consumption
- It would be nice to finalize the name of the DDS and package in this PR.
- `@fluid-experimental` was chosen as we potentially may ship this as a public api.
- `"private": true` is set in the `package.json` so **npm** will not publish those packages.

### CODE REVIEW

**Comment by anthony-murphy** [comment]
i don't really understand what this dds is, so hard to help with naming or placement. is there a design available?

**Comment by tyler-cai-microsoft** [comment]
> i don't really understand what this dds is, so hard to help with naming or placement. is there a design available?

I've updated the Task with a link to a doc.

File: packages/dds/migration-shim/package.json [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		"eslint": "~8.50.0",
+		"mocha": "^10.2.0",
+		"mocha-json-output-reporter": "^2.0.1",
+		"mocha-multi-reporters": "^1.5.1",
+		"moment": "^2.21.0",
+		"prettier": "~3.0.3",
+		"rimraf": "^4.4.0",
+		"typescript": "~5.1.6"
+	},
+	"typeValidation": {
```
Comment (ChumpChief) [inline packages/dds/migration-shim/package.json:92]: Do we still have a way to disable type validation? Ideally it should be disabled until we actually want to make a compat promise about this package.

File: packages/dds/migration-shim/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ */
+
+/**
+ * This library contains the {@link MigrationShim} distributed data structure.
+ * A `MigrationShim` is a shared object which allows a client to migrate from one data structure to another on the fly.
+ *
+ * @packageDocumentation
+ */
+
+export { MigrationShim as MigrationShim } from "./migrationShim";
```
Comment (ChumpChief) [inline packages/dds/migration-shim/src/index.ts:13]: Unnecessary alias?
```suggestion
export { MigrationShim } from "./migrationShim";
```

File: packages/dds/migration-shim/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+See the [@fluid-tools/version-tools](https://github.com/microsoft/FluidFramework/blob/main/build-tools/packages/version-tools/README.md)
+package for more information including tools to convert between version schemes.
+
+<!-- prettier-ignore-end -->
+
+<!-- AUTO-GENERATED-CONTENT:END -->
+
+## Description
+
+Experimental package - API's will change. Provides the ability to swap an existing DDS with a new DDS and migrate the
```
Comment (ChumpChief) [inline packages/dds/migration-shim/README.md:26]: Probably make this warning a little more scary (we do not want folks to experiment with it yet), and also move it to the very top (just above the auto-generated content) for maximum visibility.

**Review by ChumpChief — APPROVED** [review]
Main item is just make sure the warning is super apparent.

File: packages/dds/migration-shim/package.json (hunk 1 above)
Comment (tyler-cai-microsoft) [reply packages/dds/migration-shim/package.json:92]: I kind of just got rid of the type validation commands

Comment (tylerbutler) [reply packages/dds/migration-shim/package.json:92]: For the future, it is better to leave the scripts and add disabled: true to the typetest config. That way it's trivial to add type testing when it's needed.

**Pull request merged**